src/fswAlgorithms/_GeneralModuleFiles/kalmanFilter.h

```diff
@@ -0,0 +1,96 @@
+    void setInitialVelocity(const Eigen::VectorXd &initialVelocityInput);
+    std::optional<Eigen::VectorXd> getInitialVelocity() const;
+    void setInitialAcceleration(const Eigen::VectorXd &initialAccelerationInput);
+    std::optional<Eigen::VectorXd> getInitialAcceleration() const;
+    void setInitialBias(const Eigen::VectorXd &initialBiasInput);
+    std::optional<Eigen::VectorXd> getInitialBias() const;
+    void setInitialConsiderParameters(const Eigen::VectorXd &initialConsiderInput);
+    std::optional<Eigen::VectorXd> getInitialConsiderParameters() const;
+
+    void setInitialCovariance(const Eigen::MatrixXd &initialCovariance);
+    Eigen::MatrixXd getInitialCovariance() const;
+    void setProcessNoise(const Eigen::MatrixXd &processNoise);
+    Eigen::MatrixXd getProcessNoise() const;
+    void setUnitConversionFromSItoState(double conversion);
+    double getUnitConversionFromSItoState() const;
+    void setMeasurementNoiseScale(double measurementNoiseScale);
+    double getMeasurementNoiseScale() const;
+    void setFilterDynamics(const std::function<const StateVector(double, const StateVector&)> &dynamicsPropagator);
+
+protected:
+    virtual void customReset(){/* virtual */};
+    virtual void customInitializeUpdate(){/* virtual */};
+    virtual void customFinalizeUpdate(){/* virtual */};
+    /*! Read method neads to read incoming messages containing the measurements for the filter.
+     * Their information must be added to the Measurement container class, and added to the measurements vector.
+     * Each measurement must be paired with a measurement model provided in the measurementModels.h */
+    virtual void readFilterMeasurements(){/* virtual */};
+    virtual void writeOutputMessages(uint64_t CurrentSimNanos){/* virtual */};
+
+    virtual void timeUpdate(double updateTime)=0;
+    virtual void measurementUpdate(const MeasurementModel &measurement)=0;
+    virtual Eigen::VectorXd computeResiduals(const MeasurementModel &measurement)=0;
+    void orderMeasurementsChronologically();
+
+    std::array<std::optional<MeasurementModel>, MAX_MEASUREMENT_NUMBER> measurements;  //!< [Measurements] All
+    //!< measurement containers in chronological order
+    DynamicsModel dynamics;
+    double previousFilterTimeTag = 0; //!< [s]  Time tag for state covar/etc
+    double unitConversion = 1; //!< [-] Scale that converts input units (SI) to a desired unit for the inner maths
+    double measNoiseScaling = 1; //!< [-] Scale factor for the measurement noise
+
+    StateVector state; //!< [-] State estimate for time TimeTag
+    StateVector stateLogged; //!< [-] State variable for logging
+    Eigen::MatrixXd covar; //!< [-] covariance
+    StateVector xBar; //!< [-] Current mean state estimate
+    Eigen::VectorXd stateError; //!< [-] Current mean state error
+    Eigen::MatrixXd processNoise; //!< [-] process noise matrix
+    StateVector stateInitial; //!< [-] State estimate for time TimeTag at previous time
```

Comment:
:large_blue_circle: could group this a few lines up with the other StateVector variables. Only if you want.

Reply:
done